frontend/src/metabase-lib/expressions/typeinferencer.js

```diff
@@ -34,6 +35,8 @@ export function infer(mbql, env) {
   }
 
   switch (op) {
+    case "now":
+      return MONOTYPE.DateTime;
```

Comment:
Do you need this type specialization? Ideally a function like `now` (just like many other functions, but unlike `case`, `coalesce`) has its return type defined in `MBQL_CLAUSES` (see `frontend/src/metabase/lib/expressions/config.js`), which will get picked up by this type inferencer anyway.

Reply:
> Do you need this type specialization?

I asked that here https://github.com/metabase/metabase/pull/26548#issuecomment-1319082590
Without this change, the type of a now() expression will be inferred to be of type text, and you wouldn't be able to do things like filter on it as if it were a datetime field.

But yes, now I see that `config.js` is the ideal place to do this. I'll move this logic there.

cc @gusaiani who is working on a related issue https://github.com/metabase/metabase/issues/26512

Reply:
@ariya I've implemented your suggestion in this commit https://github.com/metabase/metabase/pull/26548/commits/1cbff0013018a5f2672e091294c22cfe751ec878

Reply:
Oh that makes sense now! Seems that we must have #26512 anyway to avoid workarounds (such as in PR #25694) that still treats `datetime` type as the generic `expression`.

From what I understand, the order should be #26512 first. If you merge this PR as is, it's still gonnna miss various type validations since no other datetime-related function adopts the special type `MONOTYPE.DateTime` yet.
Am I missing something?

Reply:
I think [this](https://github.com/metabase/metabase/commit/1cbff0013018a5f2672e091294c22cfe751ec878#diff-af943b256f8114d573abe8d77d78379b522998448998858538a2328a0ed018b7R45) line takes care of the fact that `expression` arg type can be compatible with `now`, which has type `datetime`.